e2e/smartvol_ops_test.go

```diff
@@ -407,6 +407,83 @@ func testSmartVolumeDistributeDisperse(t *testing.T) {
 	checkZeroLvs(r)
 }
 
+func testSmartVolumeAutoDistributeReplicate(t *testing.T) {
+	r := require.New(t)
+
+	smartvolname := formatVolName(t.Name())
+
+	// Too small value for max-brick-size
+	createReq := api.VolCreateReq{
+		Name:               smartvolname,
+		Size:               40 * gutils.MiB,
+		ReplicaCount:       3,
+		MaxBrickSize:       10 * gutils.MiB,
+		SubvolZonesOverlap: true,
+	}
+	volinfo, err := client.VolumeCreate(createReq)
+	r.NotNil(err)
+
+	createReq = api.VolCreateReq{
+		Name:               smartvolname,
+		Size:               40 * gutils.MiB,
+		ReplicaCount:       3,
+		MaxBrickSize:       20 * gutils.MiB,
+		SubvolZonesOverlap: true,
+	}
+	volinfo, err = client.VolumeCreate(createReq)
+	r.Nil(err)
+
+	r.Len(volinfo.Subvols, 2)
+	r.Equal("Distributed-Replicate", volinfo.Type.String())
+	r.Len(volinfo.Subvols[0].Bricks, 3)
+	r.Len(volinfo.Subvols[1].Bricks, 3)
+
+	r.Nil(client.VolumeDelete(smartvolname))
+	checkZeroLvs(r)
+
+	// Max-brick-size is more than request size
+	createReq = api.VolCreateReq{
+		Name:               smartvolname,
+		Size:               20 * gutils.MiB,
+		ReplicaCount:       3,
+		MaxBrickSize:       30 * gutils.MiB,
+		SubvolZonesOverlap: true,
+	}
+	volinfo, err = client.VolumeCreate(createReq)
+	r.Nil(err)
```

Comment:
Shouldn't this be r.NotNil (err) , how is this going to succeed if the MaxBrickSize is greater than Size ?

Reply:
`MaxBrickSize > VolumeSize` condition will always satisfy. So this satisfies user's condition.